File: January/06_Minimum_Number_of_Operations_to_Move_All_Balls_to_Each_Box/ShaFeiii.cpp
#include <bits/stdc++.h>
using namespace std;


// c++ Solution
class Solution {
public:
    vector<int> minOperations(string boxes) {
        int n = (int)boxes.size();
        vector<int> ans(n, 0);
        int ballsLeft = 0, ballsRight = 0, movesLeft = 0, movesRight = 0;
        for (int i = 0; i < n; ++i) {
            ans[i] += movesLeft;
            ballsLeft += (boxes[i] == '1');
            movesLeft += ballsLeft;
            ans[n - i - 1] += movesRight;
            ballsRight += (boxes[n - i - 1] == '1');
            movesRight += ballsRight;
        }
        return ans;
    }
};

int main() {

    return 0;
}

File: January/06_Minimum_Number_of_Operations_to_Move_All_Balls_to_Each_Box/ShaFeiii.py
class Solution(object):
    def minOperations(self, boxes):
        """
        :type boxes: str
        :rtype: List[int]
        """
        n = len(boxes)
        ans = []
        for i in range(n):
            left ,right, cnt = 0, 0, 0
            j ,k = i - 1, i + 1
            while j >= 0:
                cnt += 1
                left += (cnt if boxes[j] == '1' else 0)
                j -= 1
            cnt = 0
            while k < n:
                cnt += 1
                right += (cnt if boxes[k] == '1' else 0)
                k += 1
            ans.append(right + left)
        return ans
